File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples-python/CheckNameAvailabilityWorkspaceAvailable.py
from azure.identity import DefaultAzureCredential
from azure.mgmt.synapse import SynapseManagementClient

"""
# PREREQUISITES
    pip install azure-identity
    pip install azure-mgmt-synapse
# USAGE
    python check_name_availability_workspace_available.py

    Before run the sample, please set the values of the client ID, tenant ID and client secret
    of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID,
    AZURE_CLIENT_SECRET. For more info about how to get the value, please see:
    https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal
"""


def main():
    client = SynapseManagementClient(
        credential=DefaultAzureCredential(),
        subscription_id="00000000-1111-2222-3333-444444444444",
    )

    response = client.operations.check_name_availability(
        request={"name": "workspace1", "type": "Microsoft.ProjectArcadia/workspaces"},
    )
    print(response)


# x-ms-original-file: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/CheckNameAvailabilityWorkspaceAvailable.json
if __name__ == "__main__":
    main()

File: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples-js/CheckNameAvailabilityWorkspaceAvailable.js
const { SynapseManagementClient } = require("@azure/arm-synapse");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Check whether a workspace name is available
 *
 * @summary Check whether a workspace name is available
 * x-ms-original-file: specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/examples/CheckNameAvailabilityWorkspaceAvailable.json
 */
async function checkForAWorkspaceNameThatIsAvailable() {
  const subscriptionId = "00000000-1111-2222-3333-444444444444";
  const request = {
    name: "workspace1",
    type: "Microsoft.ProjectArcadia/workspaces",
  };
  const credential = new DefaultAzureCredential();
  const client = new SynapseManagementClient(credential, subscriptionId);
  const result = await client.operations.checkNameAvailability(request);
  console.log(result);
}

checkForAWorkspaceNameThatIsAvailable().catch(console.error);
